crypto/evp/evp_fetch.c

```diff
@@ -294,9 +294,25 @@ void *evp_generic_fetch(OPENSSL_CTX *libctx, int operation_id,
                         int (*up_ref_method)(void *),
                         void (*free_method)(void *))
 {
-    return inner_evp_generic_fetch(libctx,
-                                   operation_id, 0, name, properties,
-                                   new_method, up_ref_method, free_method);
+    void *ret = inner_evp_generic_fetch(libctx,
+                                        operation_id, 0, name, properties,
+                                        new_method, up_ref_method, free_method);
+
+    if (ret == NULL) {
+        EVPerr(0, EVP_R_FETCH_FAILED);
+#ifndef FIPS_MODE
+        ERR_add_error_data(6,
+                           openssl_ctx_is_default(libctx)
+                           ? "Default library context"
+                           : "Non-default library context",
+                           ", Algorithm (",
+                           name == NULL ? "<null>" : name,
+                           "), Properties (",
+                           properties == NULL ? "<null>" : properties,
+                           ")");
+#endif
```

Comment:
``` C
        int code = EVP_R_FETCH_FAILED;

#ifdef FIPS_MODE
        ERR_raise(ERR_LIB_EVP, code);
#else
        ERR_raise_data(ERR_LIB_EVP, code,
                       "%s, Algorithm (%s), Properties (%s)",
                       (openssl_ctx_is_default(libctx)
                        ? "Default library context"
                        : "Non-default library context"),
                       name = NULL ? "<null>" : name,
                       properties == NULL ? "<null>" : properties);
#endif
```

Dunno about you, but I find that a tad more readable...

Reply:
Ok - I changed it to that.